### ISSUE TITLE
Navigation issue / Back button on Android / Onsen+React.js

### ISSUE THREAD

**Issue by magickasoft** [OnsenUI/OnsenUI#1738]
__Environment__ 

```
[Core]
    Onsen UI 2.0.3

[Framework]
    React 15.4.1

[Framework binding]
    react-onsenui 1.0.3

[Additional libraries]
    google-map-react ^0.21.5
    i18n-zipcodes ^1.0.1
    immutable ^3.8.1
    isomorphic-fetch ^2.2.1
    promise ^7.1.1
    react-addons-shallow-compare ^15.3.2
    react-controllables ^0.6.0
    react-datepicker ^0.33.0
    react-dom ^15.4.1
    react-facebook-login ^3.4.0
    react-google-recaptcha ^0.5.4
    react-pure-render ^1.0.2
    react-redux ^4.4.5
    react-select ^1.0.0-rc.2
    react-zipcode ^1.0.0
    redux ^3.6.0
    redux-logger ^2.6.1
    redux-thunk ^2.1.0
    stylus ^0.54.5

[Platform]
    Mobile - Android ^4.1

[Browser]
    Mobile - Android WebView

```

__Encountered problem__
Experiencing a similar problem that was decribed and solved in #78 but now it seems to be back. Android Back button minimizes an application instead of navigating to a level above.

I've made two blank projects: Ionic and Onsen, set up routing and transitions between scenes. Then I compared how they work built and installed on a device in terms of Back button.
- Ionic app has Back button working both in tab bar and as we move between scenes. So we can drill-down into the app and then get back to index page with Back button; after that tap on Back minimizes the application.
- In Onsen app when I drill down into the app then tap Back it minimized the application right away. I implemented history of app's state through `windows.history` (works well in browser), and in mobile app the behavior stands still - the app is minimized, yet Back is triggered. 

__How to reproduce__

http://pastebin.com/EWi3rFrA

**Comment by asial-matagawa** [comment]
@magickasoft 
Thank you for reporting!
We have #591 as a similar issue(but too old).

We are focusing on quality improvement rather than other tasks, so we will investigate this soon.

**Comment by frandiox** [comment]
This should be fixed in core v2.2.0 and #1855
